### ISSUE TITLE
BWL Plugin timers off by up to 5-10 seconds

### ISSUE THREAD

**Issue by SapQT** [DeadlyBossMods/DeadlyBossMods#154]
Greetings,

After clearing BWL last night, it was noticed that many of the BWL timers were off by a pretty large margin. Anywhere from 5-10 seconds in most cases. This was especially noticeable on fights that required tank swaps or fights like Chromag with breath timers etc.

**Comment by MysticalOS** [comment]
"many of" but only give one example. please list every timer you want me to review that was off. I'll look at the breath timers

**Comment by MysticalOS** [comment]
I just reviewed chromagus and found 0 issues with breath timers.

First breath is 30 seconds into fight second is 60 seconds.

and every breath is 60 seconds after previous breath of that type, so everything is same as way mod is.

Here is an example log
https://classic.warcraftlogs.com/reports/8MT41VhCAdPFWNvX#fight=20&view=events&pins=2%24Off%24%23244F4B%24expression%24(ability.id%20%3D%2023309%20or%20ability.id%20%3D%2023313%20or%20ability.id%20%3D%2023189%20or%20ability.id%20%3D%2023316%20or%20ability.id%20%3D%2023312)%20and%20type%20%3D%20%22begincast%22

**Comment by psykosik** [comment]
My timers seemed accurate. The only thing I noticed but I could be wrong, on Nefarian there was no countdown on the first class call. All the subsequent calls had proper countdowns to them but the first, unless I completely missed it, was absent.

**Comment by MysticalOS** [comment]
I'll look to see why first class call is missing and addd it if consistent

**Comment by Road-block** [comment]
There is a kind of priority on abilities used by a few of the BWL bosses.
Examples:
- Firemaw will use Wing Buffet on cooldown **except** if it would coincide with Shadow Flame timer. If so Wing Buffet is deferred.
- Nefarian will use his Fear on cooldown **except** if it would coincide with Shadow Flame or class call, it is then deferred.

There might be more but those are the first ones that come to mind. This often explains the apparent variability of some timers.

Unfortunately I deleted my first week logs after uploading but I'll remember to keep them this reset.

**Comment by MysticalOS** [comment]
All of the drakes have a wing buffet timer, and nef has a fear timer too.

**Comment by Road-block** [comment]
I wasn't disputing DBM has the timers or even that they're incorrect, I was trying to say that the apparent variability of some (where wing buffet or fear doesn't come on cooldown for example) is because Blizz has a system where if 2 separate ability timers coincide one gets priority and the other gets deferred.

Sorry for the confusion.

**Comment by MysticalOS** [comment]
Yeah, spell queuing as we like to call it, has been around for 15 years. and it's much worse on retail than classic. when a boss has like 2 abilities it's no big deal. whent hey have like 9 abilities, it's a damn nightmare. Classic is easy by comparison.

But also in classic many timers aren't spell queueing issues, just very random. in many cases blizzard likes large standard variations because they like to add some randomness to encounters and don't want things to be so predictable you can pre plan abilities. Some timers for abilities have HUGE variations as a result. Other times just hidden conditions like princess, which is hard to do timers for because she uses diff abilities based on number in or out of melee. a warlock kiting her in circles solo would have TOTALLY diff timers than a group of 5 melee for example. but anyways yeah. Just look at comments I added to some of bosses for my findings on timers.
